Allow callable in dehydrate_method

TheRealVizard opened the pull request:
**Problem**

Can't reuse a `dehydrate` function across multiple Fields

**Solution**

Allow a callable to be set as the `dehydrate_method`

**Acceptance Criteria**

Run several local tests.

matthewhegarty reviewed (commented):
Looks like a good addition, thanks.

Should we add a small note to the documentation here:

https://django-import-export.readthedocs.io/en/latest/advanced_usage.html#advanced-data-manipulation-on-export

matthewhegarty commented on tests/core/tests/test_fields.py:
```diff
@@ -110,6 +110,14 @@ def testget_dehydrate_method_with_custom_method_name(self):
         method_name = field.get_dehydrate_method(resource_field_name)
         self.assertEqual(method_name, custom_dehydrate_method)
 
+    def testget_dehydrate_method_with_callable(self):
```
should the test be named `test_*` for consistency?

TheRealVizard replied:
yes my bad fixed in [60de1ef](https://github.com/django-import-export/django-import-export/pull/1950/commits/60de1efd887853ffc02b7c7b79c7ca2e44a7991f)

TheRealVizard commented:
> Looks like a good addition, thanks.
> 
> Should we add a small note to the documentation here:
> 
> https://django-import-export.readthedocs.io/en/latest/advanced_usage.html#advanced-data-manipulation-on-export

Added in [33a06e0](https://github.com/django-import-export/django-import-export/pull/1950/commits/33a06e01ecdabfdf0e90d80eb1234bc458621728) and fixed example in [4dd6584](https://github.com/django-import-export/django-import-export/pull/1950/commits/4dd6584834783c9431d091742510ae472c97311f)

matthewhegarty commented on docs/advanced_usage.rst:
```diff
@@ -759,9 +759,30 @@ In this case, the export looks like this:
+    def custom_dehydrate_callable(book):
+        return f"{book.name} by {book.author.name}"
+
+    class BookResource(resources.ModelResource):
```
Looks like you have some duplication here

TheRealVizard replied:
This is to illustrate how use both using the `dehydrate_method` with a callable and with a string

matthewhegarty reviewed (changes requested):
What is the benefit of defining a callable vs using the existing `dehydrate_<field>()`?  Both take the instance as a param so is there a specific use-case for using a callable?

TheRealVizard commented:
> What is the benefit of defining a callable vs using the existing `dehydrate_<field>()`? Both take the instance as a param so is there a specific use-case for using a callable?

Allowing the use of callables makes it easier to apply a single dehydrate function across multiple Resource classes that cannot inherit from each other or across different fields. This provides greater flexibility.

Additionally, this change allows the use of lambdas for simple transformations or calculations, making it even more convenient to apply quick, inline modifications without the need to define separate methods.

matthewhegarty reviewed (approved):
Thanks for submitting

matthewhegarty reviewed (approved):
Thanks for submitting
